lib/src/commit.rs

```diff
@@ -274,9 +274,16 @@ impl Commit {
                 resource.remove_propval(prop);
 
                 if update_index {
-                    let val = resource.get(prop)?;
-                    let atom = Atom::new(resource.get_subject().clone(), prop.into(), val.clone());
-                    remove_atoms.push(atom);
+                    if let Ok(val) = resource_unedited.get(prop) {
+                        let atom = Atom::new(resource.get_subject().clone(), prop.into(), val.clone());
```

Comment:
As another option, instead of calling `apply_changes()` twice with `update_index` set to false and true, `apply_changes()` might additionally return `remove_atoms` and `add_atoms`, and `apply_opts` would apply them.

Reply:
That's a clever idea! :)